xdsl/dialects/irdl/pyrdl_to_irdl.py

```diff
@@ -51,7 +51,7 @@ def op_def_to_irdl(op: type[IRDLOperation]) -> OperationOp:
     if result_values:
         builder.insert(ResultsOp(result_values))
 
-    return OperationOp(op_def.name, Region([block]))
+    return OperationOp(op_def.name.split(".")[-1], Region([block]))
```

Comment:
this doesn't quite feel right, are there any tests for this? My understanding is the dialect name is everything up to the first dot, and the op name is everything after it

Reply:
We probably want to centralise this logic somewhere

Reply:
Yep, should be the other way around! Split on the first dot and keep whatever's after.

More concretely, some ops have nested-like names, say `scf.forall.in_parallel` : dialect still is just `scf`, op name is `forall.in_parallel`, dot including.

And yes, we probably want to centraluze it ahah

Reply:
Any idea where the centralisation should live?

Reply:
Now rebased onto #3003